src/Functions/array/arrayAUC.cpp

```diff
@@ -116,20 +125,86 @@ namespace ErrorCodes
   *   which uses linear interpolation and can be too optimistic for the Precision Recall AUC metric.
   */
 
-template <bool PR>
+template <bool is_pr>
 class FunctionArrayAUC : public IFunction
 {
 public:
-    static constexpr auto name = PR ? "arrayAUCPR" : "arrayROCAUC";
-    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionArrayAUC<PR>>(); }
+    static constexpr auto name = is_pr ? "arrayPRAUC" : "arrayROCAUC";
```

Comment:
Please let's not create unnecessary confusion and backward-incompatibility by renaming the functions _again_. There was already some back and forth regarding naming a few hours before the last release (#72950) ...

Reply:
This change would be backward-compatible, as both `arrayPRAUC` and `arrayAUCPR` already exist. 

Right now, these are the two AUC functions:

- `arrayROCAUC`, with alias (`arrayAUC`)
- `arrayAUCPR`, with alias (`arrayPRAUC`)

The current changes update it so that the two AUC functions are:

Reply:
This change would be backward-compatible, as both `arrayPRAUC` and `arrayAUCPR` already exist (one as the main name and the other as an alias). The only change here is that we change which one is the main one and which one is the alias. 

Currently, these are the two AUC functions:

- `arrayROCAUC`, with alias (`arrayAUC`)
- `arrayAUCPR`, with alias (`arrayPRAUC`)

The proposed update changes it so that the two AUC functions become:

- `arrayROCAUC`, with alias (`arrayAUC`, `arrayAUCROC`)
- `arrayPRAUC`, with alias (`arrayAUCPR`)

And we have consistent naming across them both. Is this ok? Or should I keep it as it currently is.

Reply:
Please don't beat me for that, but as a matter of fact, the `arrayAUC*` functions are rather obscure and special purpose compared to other ClickHouse functions (like `parseDateTime`, `plus`, etc.). The purpose of aliases is to assign alternative names used by other databases (usually: MySQL, Postgres - which often use more arcane/shorter/cryptic naming) for the same SQL functions in ClickHouse. If you can have prove that 1. `arrayAUC` and friends exist in MySQL and Postgres as well, 2. the ClickHouse and other-DB versions are argument-by-argument compatible drop-in replacements and 3. a load of users is eager to port their SELECTs containing `arrayAUC` from MySQL/Postgres to ClickHouse but are blocked by the absence of corresponding aliases, then I am very happy to add new aliases. If that's not the case, we rather don't overcomplicate simple things and keep the existing naming.